flink-rpc/flink-rpc-core/src/main/java/org/apache/flink/runtime/rpc/RpcEndpoint.java

```diff
@@ -526,14 +553,21 @@ public void execute(@Nonnull Runnable command) {
         public ScheduledFuture<?> schedule(Runnable command, long delay, TimeUnit unit) {
             final long delayMillis = TimeUnit.MILLISECONDS.convert(delay, unit);
             FutureTask<Void> ft = new FutureTask<>(command, null);
-            if (mainScheduledExecutor.isShutdown()) {
-                log.warn(
-                        "The scheduled executor service is shutdown and ignores the command {}",
-                        command);
-            } else {
-                mainScheduledExecutor.schedule(
-                        () -> gateway.runAsync(ft), delayMillis, TimeUnit.MILLISECONDS);
-            }
+            getRunningFuture
+                    .get()
+                    .thenAccept(
+                            ignore -> {
+                                if (mainScheduledExecutor.isShutdown()) {
+                                    log.warn(
+                                            "The scheduled executor service is shutdown and ignores the command {}",
+                                            command);
+                                } else {
+                                    mainScheduledExecutor.schedule(
```

Comment:
Could we add an `isRunning()` method to the gateway and logs a waring if it's not running, instead of introducing `getRunningFuture`? I'm concerned that `getRunningFuture` would break some expected behaviors.